framework/opt/maybe.go

```diff
@@ -49,6 +50,21 @@ func (m Maybe[V]) OrElse(valueIfUndefined V) V {
 	return valueIfUndefined
 }
 
+// String returns a string representation of the value, or "[none]" if undefined. The string
+// representation of a value is either its own String() if it has such a method, or else the
+// result of fmt.Sprintf with "%v".
```

Comment:
This is important because `String()` gets called for values in test failure messages.